docs/DafnyRef/Types.md

```diff
@@ -1502,6 +1505,18 @@ constraint may not be satisfied.
 
 For more information about arrow types, see [Section 17](#sec-arrow-types).
 
+### 11.3.4. Witness clauses {#sec-witness}
+
+The declaration of a subset type permits an optional `witness` clause.
```

Comment:
I can't put it as a suggestion, but I strongly suggest you add examples:

```dafny
type ImpossibleInt = x : int | x < 0 && x > 0 witness *
type NonEmptySeq = x: seq<int> | |x| > 0 witness [0]
type NonNegative = x: int | |x| > 0 ghost witness 1
```
(see https://stackoverflow.com/a/49122055/1287856 for more information about ghost witnesses)